gear/form_qc_checker/src/python/form_qc_app/run.py

```diff
@@ -15,63 +20,102 @@
 
         try:
-            parameter_store = ParameterStore.create_from_environment()
-            api_key = parameter_store.get_api_key(
-                path_prefix=apikey_path_prefix)
-
-            s3_param_path = get_config(gear_context=gear_context,
+            s3_param_path = get_config(gear_context=context,
                                        key='parameter_path')
+            qc_checks_db_path = get_config(gear_context=context,
+                                           key='qc_checks_db_path')
+        except ConfigParseError as error:
+            raise GearExecutionError(
+                f'Incomplete configuration: {error.message}') from error
+
+        try:
             s3_parameters = parameter_store.get_s3_parameters(
                 param_path=s3_param_path)
 
-            qc_checks_db_path = gear_context.config.get(
-                'qc_checks_db_path', '/redcap/aws/qcchecks')
             redcap_params = parameter_store.get_redcap_report_connection(
                 param_path=qc_checks_db_path)
         except ParameterError as error:
-            log.error('Parameter error: %s', error)
-            sys.exit(1)
-        except ConfigParseError as error:
-            log.error('Incomplete configuration: %s', error.message)
-            sys.exit(1)
-
-        host = gear_context.client.api_client.configuration.host  # type: ignore
-        if api_key.split(':')[0] not in host:
-            log.error('Gearbot API key does not match host')
-            sys.exit(1)
-
-        fw_client = Client(api_key)
+            raise GearExecutionError(f'Parameter error: {error}') from error
 
         s3_client = S3BucketReader.create_from(s3_parameters)
         if not s3_client:
-            log.error('Unable to connect to S3')
-            sys.exit(1)
+            raise GearExecutionError('Unable to connect to S3')
+
+        redcap_con = REDCapReportConnection.create_from(redcap_params)
+
+        return FormQCCheckerVisitor(client=client,
+                                    file_input=file_input,
+                                    redcap_con=redcap_con,
+                                    s3_client=s3_client)
+
+    def run(self, context: GearToolkitContext):
+        """Runs the identifier lookup app.
+
+        Args:
+            context: the gear execution context
+        """
+
+        assert context, 'Gear context required'
 
-        redcap_connection = REDCapReportConnection.create_from(redcap_params)
+        run(client_wrapper=self.__client,
+            input_wrapper=self.__file_input,
+            s3_client=self.__s3_client,
+            gear_context=context,
+            redcap_connection=self.__redcap_con)
+
+
+def main():
+    """Load necessary environment variables, create Flywheel, S3 connections,
+    invoke QC app."""
 
-        run(fw_client=fw_client,
-            s3_client=s3_client,
-            gear_context=gear_context,
-            redcap_connection=redcap_connection)
+    try:
```

Comment:
GearEngine.run handles the GearExecutionError so the exception handling is not necessary. I did that to reduce the duplicated code.